### ISSUE TITLE
Node crash on Xenon

### ISSUE THREAD

**Issue by diwakergupta** [blockstack/stacks-blockchain#2270]
Version: `stacks-node 0.1.0 => 23.0.0.0 (chore/clippy:404f87d00, release build, macos [x86_64])`

Stack traces:
```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff6ab47756 __semwait_signal + 10
1   libsystem_c.dylib             	0x00007fff6aacaeea nanosleep + 196
2   stacks-node                   	0x000000010a902fa0 std::thread::sleep::h73abd6b6c0d95d1d + 80 (mod.rs:792)
3   stacks-node                   	0x0000000109d069b1 stacks_node::syncctl::PoxSyncWatchdog::pox_sync_wait::h2b9c7c607208a262 + 865 (syncctl.rs:412)
4   stacks-node                   	0x0000000109dff4a7 stacks_node::run_loop::neon::RunLoop::start::hefa0b13d881f53c9 + 8119
5   stacks-node                   	0x0000000109e4765c stacks_node::main::h437665d13b6cd0a7 + 3900
6   stacks-node                   	0x0000000109cb9dda std::sys_common::backtrace::__rust_begin_short_backtrace::h2a25c0ef8f2b7d22 + 10 (backtrace.rs:125)
7   stacks-node                   	0x0000000109df359c std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::h3f3f9c822ccc5691 + 12 (rt.rs:66)
8   stacks-node                   	0x000000010a9128b4 std::rt::lang_start_internal::h36ccce6e8a047133 + 644 (rt.rs:51)
9   stacks-node                   	0x0000000109e480d9 main + 41
10  libdyld.dylib                 	0x00007fff6aa03cc9 start + 1

Thread 1:: async-std/net
0   libsystem_kernel.dylib        	0x00007fff6ab49766 kevent + 10
1   stacks-node                   	0x000000010a71656b mio::sys::unix::kqueue::Selector::select::hfe267ac8c80feea8 + 219 (kqueue.rs:88)
2   stacks-node                   	0x000000010a713a4a mio::poll::Poll::poll1::h5f23a82125f696a1 + 858 (poll.rs:1139)
3   stacks-node                   	0x000000010a7136df mio::poll::Poll::poll::hc96778363b6f6d7f + 47 (poll.rs:1011)
4   stacks-node                   	0x0000000109e9a51f async_std::net::driver::main_loop::h56becd2504aa93b3 + 159 (mod.rs:148)
5   stacks-node                   	0x0000000109e957bd std::sys_common::backtrace::__rust_begin_short_backtrace::ha21a7b38b7eee327 + 13 (backtrace.rs:125)
6   stacks-node                   	0x0000000109e984c0 core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h5b0a56f76a95c879 + 192 (function.rs:227)
7   stacks-node                   	0x000000010a91605d std::sys::unix::thread::Thread::new::thread_start::hedb7cc0d930a8f40 + 45 (thread.rs:71)
8   libsystem_pthread.dylib       	0x00007fff6ac08109 _pthread_start + 148
9   libsystem_pthread.dylib       	0x00007fff6ac03b8b thread_start + 15

Thread 2:: chains-coordinator
0   libsystem_kernel.dylib        	0x00007fff6ab47882 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff6ac08425 _pthread_cond_wait + 698
2   stacks-node                   	0x000000010a5a0354 blockstack_lib::chainstate::coordinator::comm::CoordinatorReceivers::wait_on::h35b40c22a1e8471c + 132 (comm.rs:122)
3   stacks-node                   	0x0000000109cd6626 blockstack_lib::chainstate::coordinator::ChainsCoordinator$LT$T$C$blockstack_lib..chainstate..coordinator..comm..ArcCounterCoordinatorNotices$C$blockstack_lib..chainstate..coordinator..OnChainRewardSetProvider$GT$::run::h40e75dfa80131a68 + 710 (mod.rs:291)
4   stacks-node                   	0x0000000109cbd59c std::sys_common::backtrace::__rust_begin_short_backtrace::had4acc211d06ba8f + 268 (backtrace.rs:125)
5   stacks-node                   	0x0000000109cc4631 core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hf2a866f2fee70d75 + 241 (function.rs:227)
6   stacks-node                   	0x000000010a91605d std::sys::unix::thread::Thread::new::thread_start::hedb7cc0d930a8f40 + 45 (thread.rs:71)
7   libsystem_pthread.dylib       	0x00007fff6ac08109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6ac03b8b thread_start + 15

Thread 3:: relayer
0   libsystem_kernel.dylib        	0x00007fff6ab47882 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff6ac08425 _pthread_cond_wait + 698
2   stacks-node                   	0x000000010a903131 std::thread::park::h220cc6a203558032 + 209 (mod.rs:885)
3   stacks-node                   	0x000000010a90ef05 std::sync::mpsc::blocking::WaitToken::wait::h9e995b61a1a42bf7 + 37 (blocking.rs:63)
4   stacks-node                   	0x0000000109ca6f1d std::sync::mpsc::sync::wait::h172055e29181d14c + 141 (sync.rs:117)
5   stacks-node                   	0x0000000109c9e9b3 std::sync::mpsc::sync::Packet$LT$T$GT$::recv::hed277e7cab47f2c9 + 339 (sync.rs:280)
6   stacks-node                   	0x0000000109cbaba0 std::sys_common::backtrace::__rust_begin_short_backtrace::ha00da9fbe2013b71 + 112 (backtrace.rs:125)
7   stacks-node                   	0x0000000109cc2e47 core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h2ab55db7398bd6ac + 247 (function.rs:227)
8   stacks-node                   	0x000000010a91605d std::sys::unix::thread::Thread::new::thread_start::hedb7cc0d930a8f40 + 45 (thread.rs:71)
9   libsystem_pthread.dylib       	0x00007fff6ac08109 _pthread_start + 148
10  libsystem_pthread.dylib       	0x00007fff6ac03b8b thread_start + 15

Thread 4:: p2p
0   libsystem_kernel.dylib        	0x00007fff6ab47a46 __munmap + 10
1   stacks-node                   	0x000000010a90f344 std::sync::once::Once::call_once::_$u7b$$u7b$closure$u7d$$u7d$::h579d491f2d7021d7 + 116 (once.rs:261)
2   stacks-node                   	0x000000010a98b8d4 std::sync::once::Once::call_inner::h356a00dee8276e65 + 628 (once.rs:420)
3   stacks-node                   	0x000000010a9114d0 std::sys_common::cleanup::h7846bb8a45ec9633 + 64 (mod.rs:134)
4   stacks-node                   	0x000000010a90eddd std::process::exit::h17cf60f1a8630a8f + 13 (process.rs:1723)
5   stacks-node                   	0x0000000109e48073 stacks_node::main::_$u7b$$u7b$closure$u7d$$u7d$::ha11914e0d0e21e16 + 259 (main.rs:74)
6   stacks-node                   	0x000000010a91255e std::panicking::rust_panic_with_hook::h76436d4cf7a368ac + 350 (panicking.rs:600)
7   stacks-node                   	0x0000000109cc1e2d std::panicking::begin_panic::_$u7b$$u7b$closure$u7d$$u7d$::h5ad8119174668953 + 45
8   stacks-node                   	0x0000000109cb7e28 std::sys_common::backtrace::__rust_end_short_backtrace::ha285983fd3aa2943 + 40
9   stacks-node                   	0x000000010a94077e std::panicking::begin_panic::h9d0e31d1b59a653a + 46
10  stacks-node                   	0x0000000109cb95e2 std::sys_common::backtrace::__rust_begin_short_backtrace::h118a10c94ac54fa7 + 4626 (backtrace.rs:125)
11  stacks-node                   	0x0000000109cc3917 core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hc14182a14ab30371 + 247 (function.rs:227)
12  stacks-node                   	0x000000010a91605d std::sys::unix::thread::Thread::new::thread_start::hedb7cc0d930a8f40 + 45 (thread.rs:71)
13  libsystem_pthread.dylib       	0x00007fff6ac08109 _pthread_start + 148
14  libsystem_pthread.dylib       	0x00007fff6ac03b8b thread_start + 15

Thread 5:: dns-resolver
0   libsystem_kernel.dylib        	0x00007fff6ab47756 __semwait_signal + 10
1   libsystem_c.dylib             	0x00007fff6aacaeea nanosleep + 196
2   stacks-node                   	0x000000010a902fa0 std::thread::sleep::h73abd6b6c0d95d1d + 80 (mod.rs:792)
3   stacks-node                   	0x000000010a3b0b4c blockstack_lib::net::dns::DNSResolver::thread_main::h0deb770aaa2fa9b3 + 1580 (dns.rs:241)
4   stacks-node                   	0x0000000109cb83a3 std::sys_common::backtrace::__rust_begin_short_backtrace::h0f84aab31538771f + 35 (backtrace.rs:125)
5   stacks-node                   	0x0000000109cc33cc core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h53088878d5803b64 + 236 (function.rs:227)
6   stacks-node                   	0x000000010a91605d std::sys::unix::thread::Thread::new::thread_start::hedb7cc0d930a8f40 + 45 (thread.rs:71)
7   libsystem_pthread.dylib       	0x00007fff6ac08109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6ac03b8b thread_start + 15

```

Last few lines in the log (can attach full logs if useful, but they're a LOT)

```
DEBG [1609877181.820783] [src/net/p2p.rs:4254] [p2p] <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< End Network Dispatch <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
DEBG [1609877181.820792] [testnet/stacks-node/src/neon_node.rs:596] [p2p] P2P: backpressure: false, more downloads: true
DEBG [1609877181.820798] [testnet/stacks-node/src/neon_node.rs:609] [p2p] Atlas: attachment channel is empty
DEBG [1609877181.827223] [src/net/relay.rs:1144] [p2p] Reload read-only unconfirmed state off of 56bd4c62624b171c37cb790f7a5f5d80783806c9/7d614cec64353fff47103a5ff6908766b433fac8948395e1eb4ac1a3daec21ef
DEBG [1609877181.827314] [src/net/p2p.rs:4191] [p2p] >>>>>>>>>>>>>>>>>>>>>>> Begin Network Dispatch (poll for 100) >>>>>>>>>>>>>>>>>>>>>>>>>>>>
ERRO [1609877181.844171] [testnet/stacks-node/src/neon_node.rs:629] [p2p] P2P: Failed to process network dispatch: DBError(IndexError(SQLError(SqliteFailure(Error { code: CannotOpen, extended_code: 14 }, Some("unable to open database file")))))
Process abort due to thread panic
```

**Comment by jcnelson** [comment]
PR #2271

**Comment by diwakergupta** [comment]
Likely culprit here is that the process ran out of file descriptors. On Mac, the default is 256. The node had 182 outbound sockets for the P2P network alone:

```bash
➜ curl -s http://127.0.0.1:20443/v2/neighbors | jq '.outbound[].public_key_hash' | sort | uniq | wc -l
     182
```

**Comment by diwakergupta** [comment]
Fixed with #2271
